examples/cpp/route_guide/route_guide_callback_server.cc

```diff
@@ -201,25 +201,27 @@ class RouteGuideImpl final : public RouteGuide::CallbackService {
         StartRead(&note_);
       }
       void OnDone() override {
-        // Collect the read_starter thread if needed
-        if (read_starter_.joinable()) {
-          read_starter_.join();
-        }
         delete this;
       }
       void OnReadDone(bool ok) override {
         if (ok) {
-          // We may need to wait an arbitary amount of time on this mutex
-          // and we cannot delay the reaction, so start it in a thread
-          // Collect the previous read_starter thread if needed
-          if (read_starter_.joinable()) {
-            read_starter_.join();
-          }
-          read_starter_ = std::thread([this] {
-            mu_->lock();
-            notes_iterator_ = received_notes_->begin();
-            NextWrite();
-          });
+          // Unlike the other example in this directory that's not using
+          // the reactor pattern, we can't grab a local lock to secure the
+          // access to the notes vector, because the reactor will most likely
+          // make us jump threads, so we'll have to use a different locking
+          // strategy. We'll grab the lock locally to build a copy of the
+          // list of nodes we're going to send, then we'll grab the lock
+          // again to append the received note to the existing vector.
+          mu_->lock();
```

Comment:
Since it's now structured this way, why not a lock guard?

Reply:
BTW one of the side reasons for this example was to use a thread inside a callback since we should have some example of what to do with long blocking ops. So even if this code wasn't the right one for doing so, maybe there should be one somewhere.

Reply:
If we really wanted to shoehorn a thread there, we could instead go with a semaphore, I guess, instead of a mutex, to do the same as before? But then we're not going to be able to use a nice std construct because they've been introduced in C++20.

As for the lock guard, it may be relevant to avoid keeping the lock for the call to `NextWrite` (or, well, using a scope here... but that's maybe a bit exaggerated imho... dunno)